enhancements/authentication/AuthConfig-missing-fields.md

```diff
@@ -0,0 +1,503 @@
+A customer has a subscription for an identity provider (IdP) that they would like their employees to use for authenticating with their OpenShift clusters. The IdP provides identity services but may have certain non-standard configurations that require customization within OpenShift.
+
+#### 1. Configuring a Non-Standard OIDC Discovery URL  
+**Scenario:**  
+A company is using an external identity provider to authenticate users across multiple services, including OpenShift. However, this IdP does not follow the standard OIDC discovery URL format. To integrate OpenShift with this provider, the cluster administrator must manually specify the custom discovery URL in OpenShift’s authentication configuration.  
+
+**Steps for the Cluster Administrator:**  
+1. Update the `Authentication` CRD to include the custom `discoveryURL`:  
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+     oidcProviders:
+       - issuer:
+           discoveryURL: "https://custom-idp.example.com/.well-known/openid-configuration"
+    ```
+
+#### 2. Configuring `AudienceMatchPolicy` for Token Validation
+**Scenario:**  
+A cluster administrator needs to configure OpenShift to validate the audience (`aud`) claim in JWT tokens using a flexible matching policy.
+
+**Steps:**
+1. Update the authentication CRD with `AudienceMatchPolicy`:
+   ```yaml
+  apiVersion: config.openshift.io/v1
+  kind: Authentication
+  metadata:
+    name: cluster
+  spec:
+    type: OIDC
+    oidcProviders:
+    - name: foo
+      issuer:
+        audienceMatchPolicy: MatchAny
+     ```
+
+#### 3. Custom UID Mapping for Identity Providers
+**Scenario:**  
+A company needs to map the `uid` claim to a custom identifier in order to properly identify users across different OIDC providers.
+
+**Steps:**
+
+1. Update the authentication CRD to include the custom `uid` mapping:
+   ```yaml
+   apiVersion: config.openshift.io/v1
+    kind: Authentication
+    metadata:
+      name: cluster
+    spec:
+      type: OIDC
+      oidcProviders:
+      - name: foo
+        claimMappings:
+          uid:
+            expression: claims.email
+  ```
+-------------------------------------
+```yaml
+    apiVersion: config.openshift.io/v1
+    kind: Authentication
+    metadata:
+      name: cluster
+    spec:
+      type: OIDC
+      oidcProviders:
+      - name: foo
+        claimMappings:
+          uid:
+            claim: email
+```
+
+### 4. Adding Extra Claims for Role-Based Permissions
+**Scenario:**  
+A cluster administrator wants to utilize an external webhook authorizer to process custom claims (such as `role`) from the OIDC token for access control decisions.
+
+**Steps:**
+
+1. Update the authentication CRD to include extra claims:
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+    type: OIDC
+    oidcProviders:
+      - key: 'example.com/tenant'
```

Comment:
This goes under `.claimMappings.extra`.

Also, we should probably map this to the example of a custom `role` claim instead.